source/io/module_eggShell.js

```diff
@@ -321,6 +324,118 @@
+                    ' (dataRef: ' + valueRef.dataRef + ')');
+            }
+            var arrayBegin = Module.eggShell.dataGetArrayBegin(valueRef.dataRef);
+            var totalLength = Module.eggShell.dataGetArrayLength(valueRef.dataRef);
+            var result = Module.coreHelpers.sizedUtf8ArrayToJSString(Module.HEAPU8, arrayBegin, totalLength);
+            return result;
+        };
+
+        var findCompatibleTypedArrayConstructor = function (typeRef) {
+            var subTypeRef, isSigned, size;
+            // String will go down the Array code path a bit as is so check before array checks
+            if (Module.typeHelpers.isString(typeRef)) {
+                return Uint8Array; // exposes UTF-8 encoded array to client
+            } else if (Module.typeHelpers.isArray(typeRef)) {
+                subTypeRef = Module.typeHelpers.subElementByIndex(typeRef, 0);
+                if (Module.typeHelpers.isBoolean(subTypeRef)) {
+                    return Uint8Array;
+                } else if (Module.typeHelpers.isInteger(subTypeRef)) { // Used for Enums and Integers
+                    isSigned = Module.typeHelpers.isSigned(subTypeRef);
+                    size = Module.typeHelpers.topAQSize(subTypeRef);
+                    if (isSigned === true) {
+                        switch (size) {
+                        case 1:
+                            return Int8Array;
+                        case 2:
+                            return Int16Array;
+                        case 4:
+                            return Int32Array;
+                        default:
+                            return undefined;
+                        }
+                    } else {
+                        switch (size) {
+                        case 1:
+                            return Uint8Array;
+                        case 2:
+                            return Uint16Array;
+                        case 4:
+                            return Uint32Array;
+                        default:
+                            return undefined;
+                        }
+                    }
+                } else if (Module.typeHelpers.isFloat(subTypeRef)) {
+                    size = Module.typeHelpers.topAQSize(subTypeRef);
+                    switch (size) {
+                    case 4:
+                        return Float32Array;
+                    case 8:
+                        return Float64Array;
+                    default:
+                        return undefined;
+                    }
+                }
+            }
+            return undefined;
+        };
+
+        Module.eggShell.isTypedArrayCompatible = publicAPI.eggShell.isTypedArrayCompatible = function (valueRef) {
+            return findCompatibleTypedArrayConstructor(valueRef.typeRef) !== undefined;
+        };
+
+        Module.eggShell.getArrayDimensions = publicAPI.eggShell.getArrayDimensions = function (valueRef) {
```

Comment:
Does this mean that we are going to need another function for finding out dimensions of other types of arrays? or are we going to continue relying on `Module.eggShell.getArrayDimLength`? If any of those is the case, please consider changing the name to something that indicates that is going to be used for TypedArray compatible structures.

Reply:
Haha good catch :) I've already updated locally to just do a broader typeHelpers.isArray check instead. The `getArrayDimensions` and `resizeArray` in my local changeset for resizeArray will apply to any array.